pkg/pex/store.go

```diff
@@ -283,17 +288,18 @@ func tail(n int) func(gossipSlice) gossipSlice {
 
 	return func(gs gossipSlice) gossipSlice {
 		if n < len(gs) {
-			gs = gs[:n]
+			gs = gs[len(gs)-n:]
 		}
 
 		return gs
 	}
 }
 
-func decay(d float64) func(gossipSlice) gossipSlice {
+func decay(d float64, maxDecay int) func(gossipSlice) gossipSlice {
 	return func(gs gossipSlice) gossipSlice {
-		for len(gs) > 0 && rand.Float64() < d {
-			gs = gs[1:]
+		for len(gs) > 0 && rand.Float64() < d && maxDecay > 0 {
```

Comment:
A new random float is being generated at each iteration -- is this the desired behavior?

(I think yes, but just want to double-check.)

Reply:
Yes, yes, it is. Otherwise, the probability of evicting old entries is bigger.